File: 30 Days Of Code/day9/recursion.py
#!/bin/python3

#Day 9 of 30 days of code, Dec 17, 2022. 

import math
import os
import random
import re
import sys

def factorial(n):
    if(n == 1): #
        return 1
    else:
        n = n * factorial(n-1)
    return n
    

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input().strip())

    result = factorial(n)

    fptr.write(str(result) + '\n')

    fptr.close()

File: 30 Days Of Code/day9/recursion.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>

/*
Solution to day 9 of Hackerrank's 30 days of code.
Anton Ilic.
Dec 18 2022

*/
using namespace std;


int factorial(int n) {
    if(n != 1){
        n = n * factorial(n-1);
    }
    return n;

}

int main()
{
    int n;
    cin >> n;
    int r = factorial(n);
    cout << r << endl;
    return 0;
}

